icarusalg/Utilities/TrackTimeInterval.cxx

```diff
@@ -0,0 +1,318 @@
+/**
+ * @file icarusalg/Utilities/TrackTimeInterval.cxx
+ * @brief Utilities to constrain the time of charge detected in the TPC.
+ * @author Gianluca Petrillo (petrillo@slac.stanford.edu)
+ * @date May 18, 2023
+ * @see icarusalg/Utilities/TrackTimeInterval.h
+ */
+
+
+// library header
+#include "icarusalg/Utilities/TrackTimeInterval.h"
+
+// LArSoft libraries
+#include "larcorealg/CoreUtils/counter.h"
+#include "lardataobj/RecoBase/Hit.h"
+
+// C/C++ standard libraries
+#include <ostream>
+#include <utility> // std::move()
+#include <cmath> // std::abs()
+#include <cassert>
+
+
+// -----------------------------------------------------------------------------
+lar::util::TrackTimeInterval::TrackTimeInterval(
+  geo::GeometryCore const& geom,
+  detinfo::DetectorPropertiesData detProp,
+  detinfo::DetectorTimings detTimings
+)
+  : TrackTimeInterval
+    { buildGeomCache(geom), std::move(detProp), std::move(detTimings) }
+  {}
+
+
+// -----------------------------------------------------------------------------
+auto lar::util::TrackTimeInterval::timeRange(recob::Hit const& hit) const
+  -> TimeRange
+{
+  return timeRange(hit.PeakTime(), hit.WireID()); // that was easy...
```

Comment:
Is the conversion from `WireID` to `PlaneID` automatic? I know one can go from `WireID` to `PlaneID` of course, just didn't realize you could pass it directly.

Reply:
It is, in that a `geo::WireID` is also a `geo::PlaneID` (and `geo::TPCID` and `geo::CryostatID`) by inheritance.
You can be explicit and go `.asPlaneID()`, this time I apparently chose not to.
That also means one needs to apply some care when using functions that can support both IDs (not many in LArSoft).

Reply:
🙌